lib/grandpa/grandpa.go

```diff
@@ -1298,6 +1298,70 @@ func (s *Service) findParentWithNumber(v *Vote, n uint32) (*Vote, error) {
 	return NewVoteFromHeader(b), nil
 }
 
+// GetSetID returns the current setID
+func (s *Service) GetSetID() uint64 {
+	return s.state.setID
+}
+
+// GetRound return the current round number
+func (s *Service) GetRound() uint64 {
+	s.roundLock.Lock()
+	defer s.roundLock.Unlock()
+
+	return s.state.round
+}
+
+// GetVoters returns the list of current grandpa.Voters
+func (s *Service) GetVoters() Voters {
+	return s.state.voters
+}
+
+// PreVotes returns the current prevotes to the current round
+func (s *Service) PreVotes() []ed25519.PublicKeyBytes {
+	s.mapLock.Lock()
+	defer s.mapLock.Unlock()
+
+	votes := make([]ed25519.PublicKeyBytes, 0, s.lenVotes(prevote)+len(s.pvEquivocations))
+
+	s.prevotes.Range(func(k interface{}, _ interface{}) bool {
+		b := k.(ed25519.PublicKeyBytes)
+
+		// if the key is not in the equivocations, then add
```

Comment:
the same key should never be in both votes and equivocations lists, so you can remove the check. same for below

Reply:
Done!

Reply:
So, I can merge `prevotes` and `pvEquivocations` without any check and will not end up with any duplicates?